test/test.hs

```diff
@@ -205,10 +205,10 @@ wrapOn_0 :: (Fork a -> IO b) -> IO b
 wrapOn_0 = wrap $ ThreadGroup.forkOn 0
 
 wrapIOWithUnmask :: (Fork a -> IO b) -> IO b
-wrapIOWithUnmask = wrap $ \tg m -> ThreadGroup.forkIOWithUnmask tg $ const m
+wrapIOWithUnmask = wrap $ \tg m -> ThreadGroup.forkIOWithUnmask tg (\_ -> m)
```

Comment:
Let me guess: `const` doesn't work any more due to a GHC type checker modification?

Reply:
Right, I think the relevant change is [simplified subsumption](https://github.com/ghc-proposals/ghc-proposals/blob/master/proposals/0287-simplify-subsumption.rst). Here is the error:

```
[1 of 2] Compiling Main             ( test/test.hs, /home/janus/flipstone/threads/dist-newstyle/build/x86_64-linux/ghc-9.4.1/threads-0.5.1.6/build/test-threads/test-threads-tmp/Main.o )

test/test.hs:208:70: error:
    • Couldn't match expected type: forall b. IO b -> IO b
                  with actual type: IO b1 -> IO b1
    • In the second argument of ‘($)’, namely ‘const m’
      In the expression: ThreadGroup.forkIOWithUnmask tg $ const m
      In the second argument of ‘($)’, namely
        ‘\ tg m -> ThreadGroup.forkIOWithUnmask tg $ const m’
    |
208 | wrapIOWithUnmask = wrap $ \tg m -> ThreadGroup.forkIOWithUnmask tg $ const m
    |                                                                      ^^^^^^^
```